diff --unidirectional-new-file -ru -x 'PortIndex*' ports.orig/archivers/xz/Portfile ports/archivers/xz/Portfile
--- ports.orig/archivers/xz/Portfile	2015-10-02 06:01:50.000000000 -0700
+++ ports/archivers/xz/Portfile	2016-03-15 11:34:34.000000000 -0700
@@ -26,9 +26,9 @@
 checksums       rmd160  6c376338dbff31950603738b8ed5ba129b9dcf7c \
                 sha256  6ff5f57a4b9167155e35e6da8b529de69270efb2b4cf3fbabf41a4ee793840b5
 
-depends_lib     port:libiconv port:gettext
+depends_lib     port:gettext
 
-configure.args  --with-libiconv-prefix=${prefix} --with-libintl-prefix=${prefix}
+configure.args  --with-libintl-prefix=${prefix}
 
 # document that we always need legacy symlinks now that "lzmautils" refers here
 configure.args-append --enable-lzma-links
diff --unidirectional-new-file -ru -x 'PortIndex*' ports.orig/devel/apr-util/Portfile ports/devel/apr-util/Portfile
--- ports.orig/devel/apr-util/Portfile	2014-09-29 09:30:23.000000000 -0700
+++ ports/devel/apr-util/Portfile	2016-03-15 11:34:34.000000000 -0700
@@ -21,13 +21,13 @@
 		rmd160	9710ffbffea60750b2c1f75532c607bda87ecb71
 
 depends_lib	port:apr port:expat \
-		port:libiconv port:db46 \
+		port:db46 \
 		port:sqlite3
 
 use_parallel_build	yes
 configure.ccache	no
 configure.args	--with-apr=${prefix}/bin/apr-1-config --with-expat=${prefix} \
-		--with-iconv=${prefix} --with-berkeley-db=${prefix}/include:${prefix}/lib/db46 \
+		--with-berkeley-db=${prefix}/include:${prefix}/lib/db46 \
 		--without-mysql --without-pgsql --without-sqlite2 --with-sqlite3=${prefix}
 
 test.run	yes
diff --unidirectional-new-file -ru -x 'PortIndex*' ports.orig/devel/atk/Portfile ports/devel/atk/Portfile
--- ports.orig/devel/atk/Portfile	2015-10-04 12:30:49.000000000 -0700
+++ ports/devel/atk/Portfile	2016-03-15 11:34:34.000000000 -0700
@@ -28,8 +28,7 @@
 depends_build       port:pkgconfig
 
 depends_lib         path:lib/pkgconfig/glib-2.0.pc:glib2 \
-                    port:gettext \
-                    port:libiconv
+                    port:gettext
 
 gobject_introspection yes
 
diff --unidirectional-new-file -ru -x 'PortIndex*' ports.orig/devel/gettext/Portfile ports/devel/gettext/Portfile
--- ports.orig/devel/gettext/Portfile	2015-12-31 10:01:15.000000000 -0800
+++ ports/devel/gettext/Portfile	2016-03-15 11:34:34.000000000 -0700
@@ -51,7 +51,6 @@
                         --without-xz
 
 depends_lib             port:expat \
-                        port:libiconv \
                         port:ncurses
 
 test.run                yes
diff --unidirectional-new-file -ru -x 'PortIndex*' ports.orig/devel/glib2/Portfile ports/devel/glib2/Portfile
--- ports.orig/devel/glib2/Portfile	2015-11-28 20:01:03.000000000 -0800
+++ ports/devel/glib2/Portfile	2016-03-15 11:34:34.000000000 -0700
@@ -56,7 +56,6 @@
 
 depends_lib                 port:gettext \
                             port:libffi \
-                            port:libiconv \
                             port:pcre \
                             port:zlib
 
diff --unidirectional-new-file -ru -x 'PortIndex*' ports.orig/devel/glib2/work/.macports.glib2.state ports/devel/glib2/work/.macports.glib2.state
--- ports.orig/devel/glib2/work/.macports.glib2.state	1969-12-31 16:00:00.000000000 -0800
+++ ports/devel/glib2/work/.macports.glib2.state	2016-03-15 11:34:34.000000000 -0700
@@ -0,0 +1 @@
+target: org.macports.archivefetch
diff --unidirectional-new-file -ru -x 'PortIndex*' ports.orig/editors/vim/Portfile ports/editors/vim/Portfile
--- ports.orig/editors/vim/Portfile	2016-01-18 03:01:30.000000000 -0800
+++ ports/editors/vim/Portfile	2016-03-15 11:34:34.000000000 -0700
@@ -23,8 +23,7 @@
 
 depends_build       bin:grep:grep
 depends_lib         port:ncurses \
-                    port:gettext \
-                    port:libiconv
+                    port:gettext
 
 patchfiles-append patch-compat-sigaltstack.diff
 
diff --unidirectional-new-file -ru -x 'PortIndex*' ports.orig/gnome/gtk2/Portfile ports/gnome/gtk2/Portfile
--- ports.orig/gnome/gtk2/Portfile	2016-03-06 04:00:17.000000000 -0800
+++ ports/gnome/gtk2/Portfile	2016-03-15 11:34:34.000000000 -0700
@@ -45,8 +45,13 @@
 # indirectly depends on gtk2, so we can not list it as a dependency here.
 #                    port:librsvg
 
-depends_run         port:shared-mime-info \
-                    port:hicolor-icon-theme
+# If shared-mime-info is installed it causes GTK+ to use Glib's GIO to
+# do automatic file type guessing and to disable its own internal
+# guessing. The guessing then fails at runtime if the MIME database
+# (/usr/share/mime) isn't present. Avoid using GIO and always use GTK+'s
+# internal guesser.
+# depends_run         port:shared-mime-info
+depends_run         port:hicolor-icon-theme
 
 patchfiles-append   patch-aliases.diff
 patchfiles-append   patch-gtk-builder-convert.diff
@@ -55,7 +60,12 @@
 use_autoreconf      yes
 autoreconf.args     -fvi
 
+# Don't have the immodules as loadable modules. This would require
+# running install_name_tool on the .so files to make the libraries they
+# depend on relative to @executable_path, not the MacPorts installation
+# directory.
 configure.args      --enable-static \
+                    --with-included-immodules \
                     --disable-glibtest \
                     --enable-introspection \
                     gio_can_sniff=yes
@@ -100,21 +110,6 @@
     xinstall -d ${destroot}${docdir}
     xinstall -m 644 -W ${worksrcpath} AUTHORS COPYING HACKING NEWS README \
         ${destroot}${docdir}
-
-    ui_debug "Creating gtk.immodules..."
-    system "DYLD_LIBRARY_PATH=${destroot}${prefix}/lib \
-        ${destroot}${prefix}/bin/gtk-query-immodules-2.0 \
-        ${destroot}${prefix}/lib/gtk-2.0/2.10.0/immodules/*.so \
-        >${destroot}${prefix}/etc/gtk-2.0/gtk.immodules"
-    reinplace "s|${destroot}||" ${destroot}${prefix}/etc/gtk-2.0/gtk.immodules
-}
-
-post-activate {
-    ui_debug "Updating gdk-pixbuf.loaders..."
-    system "${prefix}/bin/gdk-pixbuf-query-loaders \
-        >${prefix}/etc/gtk-2.0/gdk-pixbuf.loaders"
-    ui_debug "Updating immodules.cache..."
-    system "${prefix}/bin/gtk-query-immodules-2.0 --update-cache"
 }
 
 platform darwin 8 {
diff --unidirectional-new-file -ru -x 'PortIndex*' ports.orig/graphics/fontconfig/Portfile ports/graphics/fontconfig/Portfile
--- ports.orig/graphics/fontconfig/Portfile	2014-04-07 20:00:05.000000000 -0700
+++ ports/graphics/fontconfig/Portfile	2016-03-15 11:34:34.000000000 -0700
@@ -26,8 +26,7 @@
 
 depends_build               port:pkgconfig
 
-depends_lib                 port:libiconv \
-                            port:expat \
+depends_lib                 port:expat \
                             port:freetype
 
 if {${os.platform} eq "darwin" && ${os.major} < 9} {
diff --unidirectional-new-file -ru -x 'PortIndex*' ports.orig/graphics/gdk-pixbuf2/Portfile ports/graphics/gdk-pixbuf2/Portfile
--- ports.orig/graphics/gdk-pixbuf2/Portfile	2015-12-14 14:30:29.000000000 -0800
+++ ports/graphics/gdk-pixbuf2/Portfile	2016-03-15 11:34:34.000000000 -0700
@@ -29,7 +29,6 @@
 depends_build       port:pkgconfig
 
 depends_lib         path:lib/pkgconfig/glib-2.0.pc:glib2 \
-                    port:shared-mime-info \
                     port:libpng \
                     port:jasper \
                     port:jpeg \
@@ -42,7 +41,15 @@
 use_autoreconf      yes
 autoreconf.args     -fvi
 
+# Don't have the GDK image loaders as loadable modules. This would
+# require running install_name_tool on the .so files to make the
+# libraries they depend on relative to @executable_path, not the
+# MacPorts installation directory.
+# See also https://trac.macports.org/ticket/45354
 configure.args      --with-libjasper \
+                    --disable-modules \
+                    --disable-gio-sniffing \
+                    --with-included-loaders=yes \
                     --disable-silent-rules
 
 test.run            yes
diff --unidirectional-new-file -ru -x 'PortIndex*' ports.orig/mail/libidn/Portfile ports/mail/libidn/Portfile
--- ports.orig/mail/libidn/Portfile	2015-08-28 11:01:26.000000000 -0700
+++ ports/mail/libidn/Portfile	2016-03-15 11:34:34.000000000 -0700
@@ -25,7 +25,7 @@
 checksums           	rmd160  35564ab88310df83d0527c10ca07850a9219b86a \
                     	sha256  ba5d5afee2beff703a34ee094668da5c6ea5afa38784cebba8924105e185c4f5
 
-depends_lib		port:libiconv port:gettext
+depends_lib		port:gettext
 
 patchfiles		configure-no-nawk.patch
 post-patch {
diff --unidirectional-new-file -ru -x 'PortIndex*' ports.orig/perl/p5-locale-gettext/Portfile ports/perl/p5-locale-gettext/Portfile
--- ports.orig/perl/p5-locale-gettext/Portfile	2016-01-12 14:01:24.000000000 -0800
+++ ports/perl/p5-locale-gettext/Portfile	2016-03-15 11:35:28.000000000 -0700
@@ -24,8 +24,7 @@
 
 if {${perl5.major} != ""} {
     depends_lib-append \
-                    port:gettext \
-                    port:libiconv
+                    port:gettext
 
     patchfiles      patch-Makefile.PL
 
diff --unidirectional-new-file -ru -x 'PortIndex*' ports.orig/python/py-libxml2/files/patch-setup.py.diff ports/python/py-libxml2/files/patch-setup.py.diff
--- ports.orig/python/py-libxml2/files/patch-setup.py.diff	2014-10-20 22:05:02.000000000 -0700
+++ ports/python/py-libxml2/files/patch-setup.py.diff	2016-03-15 11:34:35.000000000 -0700
@@ -13,8 +13,20 @@
  ];
  
  xml_includes=""
-@@ -85,7 +81,7 @@
+@@ -73,19 +69,9 @@
+     print("failed to find headers for libxml2: update includes_dir")
+     sys.exit(1)
  
+-iconv_includes=""
+-for dir in includes_dir:
+-    if not missing(dir + "/iconv.h"):
+-        iconv_includes=dir
+-	break;
+-
+-if iconv_includes == "":
+-    print("failed to find headers for libiconv: update includes_dir")
+-    sys.exit(1)
+-
  # those are added in the linker search path for libraries
  libdirs = [
 -os.path.join(ROOT,'lib'),
@@ -55,3 +67,12 @@
  
  if with_xslt == 1:
      xslt_includes=""
+@@ -166,7 +126,7 @@
+ if WITHDLLS:
+     modules.append('libxmlmods.__init__')
+ c_files = ['libxml2-py.c', 'libxml.c', 'types.c' ]
+-includes= [xml_includes, iconv_includes]
++includes= [xml_includes]
+ libs    = [libraryPrefix + "xml2"] + platformLibs
+ macros  = []
+ if with_threads:
diff --unidirectional-new-file -ru -x 'PortIndex*' ports.orig/textproc/help2man/Portfile ports/textproc/help2man/Portfile
--- ports.orig/textproc/help2man/Portfile	2015-11-23 19:00:57.000000000 -0800
+++ ports/textproc/help2man/Portfile	2016-03-15 11:34:35.000000000 -0700
@@ -28,7 +28,7 @@
 set pbranch     5.22
 
 depends_lib     port:perl${pbranch} port:p${pbranch}-locale-gettext \
-                port:gettext port:libiconv
+                port:gettext
 
 configure.perl  ${prefix}/bin/perl${pbranch}
 
diff --unidirectional-new-file -ru -x 'PortIndex*' ports.orig/textproc/libxml2/Portfile ports/textproc/libxml2/Portfile
--- ports.orig/textproc/libxml2/Portfile	2015-03-10 04:00:55.000000000 -0700
+++ ports/textproc/libxml2/Portfile	2016-03-15 11:34:35.000000000 -0700
@@ -20,8 +20,7 @@
                     available in other environments.
 homepage            http://xmlsoft.org/
 
-depends_lib         port:libiconv \
-                    port:xz \
+depends_lib         port:xz \
                     port:zlib
 
 master_sites        ftp://gd.tuwien.ac.at/pub/libxml/ \
diff --unidirectional-new-file -ru -x 'PortIndex*' ports.orig/textproc/libxslt/Portfile ports/textproc/libxslt/Portfile
--- ports.orig/textproc/libxslt/Portfile	2015-06-03 12:01:19.000000000 -0700
+++ ports/textproc/libxslt/Portfile	2016-03-15 11:34:35.000000000 -0700
@@ -38,8 +38,7 @@
 if {${name} eq ${subport}} {
     revision            1
 
-    depends_lib-append  port:libiconv \
-                        port:libxml2 \
+    depends_lib-append  port:libxml2 \
                         port:zlib
 
 pre-configure {
diff --unidirectional-new-file -ru -x 'PortIndex*' ports.orig/x11/pango/Portfile ports/x11/pango/Portfile
--- ports.orig/x11/pango/Portfile	2015-10-14 14:32:28.000000000 -0700
+++ ports/x11/pango/Portfile	2016-03-15 11:34:35.000000000 -0700
@@ -46,6 +46,7 @@
 patchfiles              patch-ltmain.sh.diff
 
 configure.args          --enable-static \
+                        --with-included-modules \
                         --disable-silent-rules \
                         --without-x
 
