From 4523e52bb5c54f4d65cd3672e8f875c137b44f77 Mon Sep 17 00:00:00 2001
From: Daniel Miller <daniel@bonsaiviking.com>
Date: Tue, 15 Sep 2015 13:16:39 +0000
Subject: [PATCH] Disable TPACKET_V3 in included libpcap. Still broken, see #34

---
 libpcap/pcap-linux.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/libpcap/pcap-linux.c b/libpcap/pcap-linux.c
index a226da1..a17c250 100644
--- a/libpcap/pcap-linux.c
+++ b/libpcap/pcap-linux.c
@@ -188,6 +188,10 @@
 # endif /* PACKET_HOST */
 
 
+ /* TPACKET_V3 is broken. Temporarily disabling support here until upstream
+  * libpcap addresses the issue. */
+#undef TPACKET3_HDRLEN
+
  /* check for memory mapped access avaibility. We assume every needed
   * struct is defined if the macro TPACKET_HDRLEN is defined, because it
   * uses many ring related structs and macros */
-- 
1.9.1

